packages/lib/src/lib.tsx

```diff
@@ -41,13 +42,15 @@ export const Turnstile = forwardRef<TurnstileInstance | undefined, TurnstileProp
 	const firstRendered = useRef(false)
 	const [widgetId, setWidgetId] = useState<string | undefined | null>()
 	const [turnstileLoaded, setTurnstileLoaded] = useState(false)
-	const scriptId = scriptOptions?.id || DEFAULT_SCRIPT_ID
-	const scriptLoaded = useObserveScript(scriptId)
 	const containerId = id ?? DEFAULT_CONTAINER_ID
+	const scriptId = injectScript
+		? scriptOptions?.id || `${DEFAULT_SCRIPT_ID}__${containerId}`
+		: scriptOptions?.id || DEFAULT_SCRIPT_ID
+	const scriptLoaded = useObserveScript(scriptId)
 
-	const onLoadCallbackName = `${
-		scriptOptions?.onLoadCallbackName || DEFAULT_ONLOAD_NAME
-	}#${containerId}`
+	const onLoadCallbackName = scriptOptions?.onLoadCallbackName
+		? `${scriptOptions.onLoadCallbackName}__${containerId}`
+		: `${DEFAULT_ONLOAD_NAME}__${containerId}`
```

Comment:
The `scriptId` and `onLoadCallbackName` now may include the `containerId`. So maybe we should update `DEFAULT_ONLOAD_NAME` to `DEFAULT_ONLOAD_NAME_PREFIX` and `DEFAULT_SCRIPT_ID` to `DEFAULT_SCRIPT_ID_PREFIX`.